frontend_tests/casper_tests/09-navigation.js

```diff
@@ -8,28 +8,61 @@ casper.then(function () {
     casper.test.info('Testing navigation');
 });
 
+function wait_for_tab(tab) {
+    casper.waitForSelector('#' + tab + '.tab-pane.active', function () {
+        casper.test.assertExists('#' + tab + '.tab-pane.active', tab + ' page is active');
+    });
+}
+
 function then_navigate_to(click_target, tab) {
     casper.then(function () {
         casper.test.info('Visiting #' + click_target);
         casper.click('a[href^="#' + click_target + '"]');
+        wait_for_tab(tab);
     });
+}
 
-    casper.waitForSelector('#' + tab + '.tab-pane.active', function () {
-        casper.test.assertExists('#' + tab + '.tab-pane.active', tab + ' page is active');
+function then_navigate_to_settings() {
+    casper.then(function () {
+        casper.test.info('Navigate to settings');
+        var menu_selector = '#settings-dropdown';
+        casper.waitUntilVisible(menu_selector, function () {
+            casper.click(menu_selector);
+            casper.waitUntilVisible('a[href^="#settings"]', function () {
+                casper.click('a[href^="#settings"]');
+                wait_for_tab('settings');
+            });
+        });
+    });
+}
+
+function then_navigate_to_subscriptions() {
+    casper.then(function () {
+        casper.test.info('Navigate to subscriptions');
+
+        var menu_selector = '#settings-dropdown';
+        casper.waitUntilVisible(menu_selector, function () {
+            casper.click(menu_selector);
+            casper.then(function () {
+                casper.click('a[href^="#subscriptions"]');
+                wait_for_tab('subscriptions');
+            });
+        });
     });
 }
 
 // Take a navigation tour of the app.
 // Entries are (click target, tab that should be active after clicking).
-var tabs = [["settings", "settings"], ["home", "home"],
-            ["subscriptions", "subscriptions"], ["", "home"],
-            ["settings", "settings"], ["narrow/stream/Verona", "home"],
-            ["subscriptions", "subscriptions"], ["narrow/is/private", "home"]];
-var i;
-
-for (i=0; i<tabs.length; i++) {
-    then_navigate_to(tabs[i][0], tabs[i][1]);
-}
+
+then_navigate_to_settings();
+then_navigate_to('narrow/stream/Verona', 'home');
+then_navigate_to('home', 'home');
+then_navigate_to_subscriptions();
+then_navigate_to('', 'home');
+then_navigate_to_settings();
+then_navigate_to('narrow/stream/Verona', 'home');
+then_navigate_to_subscriptions();
+then_navigate_to('narrow/is/private', 'home');
```

Comment:
This looks good!  I think we'll eventually want to move this navigation code into `common.js` and reuse it in other parts of the app :)